src/Catty/SetTempoToBrick+Instruction.swift

```diff
@@ -0,0 +1,38 @@
+/**
+ *  Copyright (C) 2010-2021 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc extension SetTempoToBrick: CBInstructionProtocol {
+
+    @nonobjc func instruction() -> CBInstruction {
+        guard let spriteObject = self.script?.object else { fatalError("This should never happen") }
+        let spriteObjectName = spriteObject.name
+
+        return CBInstruction.execClosure { context, scheduler in
+            let audioEngine = scheduler.getAudioEngine()
+            if spriteObjectName != nil {
+                audioEngine.setTempo(tempo: context.formulaInterpreter.interpretInteger(self.getFormulas()[0], for: spriteObject))
+                context.state = .runnable
```

Comment:
Changing the context's state should not be necessary and recommended for this simple Brick. Is there a special reason why you did that?